Src/CSharpier/CommandLineFormatter.cs

```diff
@@ -20,44 +19,93 @@ public class CommandLineFormatter
 
         protected readonly Stopwatch Stopwatch;
 
-        protected readonly string RootPath;
+        protected readonly string BaseDirectoryPath;
 
         protected readonly CommandLineOptions CommandLineOptions;
         protected readonly PrinterOptions PrinterOptions;
         protected readonly IFileSystem FileSystem;
+        protected readonly IConsole Console;
+        protected readonly IgnoreFile IgnoreFile;
 
-        public global::Ignore.Ignore Ignore { get; set; }
-
-        public CommandLineFormatter(
-            string rootPath,
+        protected CommandLineFormatter(
+            string baseDirectoryPath,
             CommandLineOptions commandLineOptions,
             PrinterOptions printerOptions,
-            IFileSystem fileSystem
+            IFileSystem fileSystem,
+            IConsole console,
+            IgnoreFile ignoreFile
         ) {
-            this.RootPath = rootPath;
+            this.BaseDirectoryPath = baseDirectoryPath;
             this.PrinterOptions = printerOptions;
             this.CommandLineOptions = commandLineOptions;
             this.Stopwatch = Stopwatch.StartNew();
             this.FileSystem = fileSystem;
-            this.Ignore = new global::Ignore.Ignore();
+            this.Console = console;
+            this.IgnoreFile = ignoreFile;
         }
 
-        public async Task<int> Format(CancellationToken cancellationToken)
-        {
-            var ignoreExitCode = await this.ParseIgnoreFile(cancellationToken);
-            if (ignoreExitCode != 0)
+        public static async Task<int> Format(
+            CommandLineOptions commandLineOptions,
+            IFileSystem fileSystem,
+            IConsole console,
+            CancellationToken cancellationToken
+        ) {
+            var baseDirectoryPath = File.Exists(commandLineOptions.DirectoryOrFile)
+                ? Path.GetDirectoryName(commandLineOptions.DirectoryOrFile)
+                : commandLineOptions.DirectoryOrFile;
+
+            if (baseDirectoryPath == null)
             {
-                return ignoreExitCode;
+                throw new Exception(
+                    $"The path of {commandLineOptions.DirectoryOrFile} does not appear to point to a directory or a file."
+                );
             }
 
+            var configurationFileOptions = ConfigurationFileOptions.Create(
+                baseDirectoryPath,
+                fileSystem
+            );
+
+            var (ignoreFile, exitCode) = await CSharpier.IgnoreFile.Create(
```

Comment:
So, returning a tuple with the result and a flag representing success (as an int) isn't a great pattern. 

One way would be to have multiple types (each representing a different status code, along with the result for the success case) that inherit from a common base, and return said base. Then you can pattern match on the different subtypes and proceed accordingly.

However, in this case it looks like you're only using the values `0` and `1` to represent success and failure. In this case, the nullability of the result already achieves that quite well, so the whole tuple is redundant. You can just return `1` below if the `ignoreFile` is `null`.